src/engraving/libmscore/chord.cpp

```diff
@@ -1446,33 +1446,29 @@ int Chord::calcMinStemLength()
         if (!_up && line < -2) {
             outsideStaffOffset = -line;
         } else if (_up && line > staff()->lines(tick()) * 4) {
-            outsideStaffOffset = line - staff()->lines(tick()) * 4 + 4;
+            outsideStaffOffset = line - (staff()->lines(tick()) * 4) + 4;
         }
 
         minStemLength += (outSidePadding + std::max(noteSidePadding, outsideStaffOffset));
 
         if (_hook) {
+            bool straightFlags = score()->styleB(Sid::useStraightNoteFlags);
             double smuflAnchor = _hook->smuflAnchor().y() * (_up ? 1 : -1);
-            int hookOffset = (_hook->height() + smuflAnchor) / _spatium * 4 - 2;
+            int hookOffset = floor((_hook->height() / _relativeMag + smuflAnchor) / _spatium * 4) - (straightFlags ? 0 : 2);
+            hookOffset = std::min(hookOffset, 11);
```

Comment:
You should use a named constant (or at least add a comment) explaining what this number is for

Reply:
The same issue for all the magic numbers in this PR

Reply:
Added comments in and also rebased!